### PR TITLE
[libnice] Fix build error on Linux

### LINKED ISSUE #13483: [libnice] Can't build on something else than Windows
**Host Environment**
- OS: Linux
- Compiler: latest vcpkg + gcc 7.5

**To Reproduce**
`./vcpkg install libnice`

**Failure logs** 
Logs are too big and complex to be put here. The whole build system is broken. It looks statically for Windows libs and headers.
Even after fixing the port's CMakeList.txt there are build issues.

### PR DESCRIPTION
**Pull request by LilyWangL**
**Describe the pull request**

- What does your PR fix? Fixes #13483
Fix `libnice` build failed on Linux.

### CODE REVIEW

File: ports/libnice/CMakeLists.txt [hunk 1]
```diff
@@ -121,8 +130,14 @@ SET(HEADERS
 	./win32/vs9/config.h
 	)
 add_library(libnice ${SRCS} ${HEADERS})
+if (MSVC)
 set_target_properties(${PROJECT_NAME} PROPERTIES COMPILE_FLAGS "/wd4244 /wd4005 /wd4391 /wd4142 /wd4267")
```
Comment (JackBoosY) [inline ports/libnice/CMakeLists.txt:134]:
```suggestion
    set_target_properties(${PROJECT_NAME} PROPERTIES COMPILE_FLAGS "/wd4244 /wd4005 /wd4391 /wd4142 /wd4267")
```

File: ports/libnice/CMakeLists.txt [hunk 2]
```diff
@@ -121,8 +130,14 @@ SET(HEADERS
...
+endif()
+if (WIN32)
 target_link_libraries(libnice ${IPHLPAPI_LIBRARY} ${WS2_32_LIB} ${ADVAPI32_LIB} ${GLIB_LIBRARY} ${GOBJECT_LIBRARY} ${GIO_LIBRARY} OpenSSL::SSL)
```
Comment (JackBoosY) [inline ports/libnice/CMakeLists.txt:137]:
```suggestion
    target_link_libraries(libnice ${IPHLPAPI_LIBRARY} ${WS2_32_LIB} ${ADVAPI32_LIB} ${GLIB_LIBRARY} ${GOBJECT_LIBRARY} ${GIO_LIBRARY} OpenSSL::SSL)
```

File: ports/libnice/CMakeLists.txt [hunk 3]
```diff
@@ -121,8 +130,14 @@ SET(HEADERS
...
+else()
+target_link_libraries(libnice ${GLIB_LIBRARY} ${GOBJECT_LIBRARY} ${GIO_LIBRARY} OpenSSL::SSL)
```
Comment (JackBoosY) [inline ports/libnice/CMakeLists.txt:139]:
```suggestion
    target_link_libraries(libnice ${GLIB_LIBRARY} ${GOBJECT_LIBRARY} ${GIO_LIBRARY} OpenSSL::SSL)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libnice/CMakeLists.txt [hunk 4]
```diff
@@ -4,11 +4,20 @@ project(libnice
...
+if (WIN32)
+    add_definitions(
+        -D_CRT_SECURE_NO_WARNINGS
+        -D_SCL_SECURE_NO_WARNINGS
+        -DHAVE_OPENSSL
+        -DPACKAGE_STRING="${PROJECT_VERSION}")
+else()
+    add_definitions(
+        -D_CRT_SECURE_NO_WARNINGS
+        -D_SCL_SECURE_NO_WARNINGS
```
Comment (JackBoosY) [inline ports/libnice/CMakeLists.txt:16]:
```suggestion
```

**Review by JackBoosY — APPROVED** [review]

**Comment by LilyWangL** [comment]
Baseline error will be fixed in PR #14400

**Pull request merged**